### PR TITLE
[type:refactor] Update MenuProject, MenuModule, MenuDocItem to VO objects

### PR DESCRIPTION
**Pull request by GOODBOY008**
Update MenuProject, MenuModule, MenuDocItem to VO objects

### CODE REVIEW

**Review by moremind — DISMISSED** [review]

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -94,7 +93,7 @@ jobs:
           cache: maven
 
       - name: Build shenyu-e2e-engine with Maven
```
Comment (moremind) [inline .github/workflows/e2e.yaml:95]: not change this

**Review by moremind — CHANGES_REQUESTED** [review]

File: .github/workflows/e2e.yaml (hunk 1 above)
Comment (GOODBOY008) [reply .github/workflows/e2e.yaml:95]: Just try to trigger maven cache for test

Comment (GOODBOY008) [reply .github/workflows/e2e.yaml:95]: > not change this

CI was fail, I found maven cache do not hava e2e-moudle dependency. @moremind

Comment (moremind) [reply .github/workflows/e2e.yaml:95]: yes, e2e not cache

Comment (GOODBOY008) [reply .github/workflows/e2e.yaml:95]: > yes, e2e not cache

Anybody to fix ci fail ?

Comment (moremind) [reply .github/workflows/e2e.yaml:95]: maybe ci cache is error, and @yunlongn @damonxue will fix it

Comment (moremind) [reply .github/workflows/e2e.yaml:95]: @HaiqiQin will fix it

Comment (moremind) [reply .github/workflows/e2e.yaml:95]: @GOODBOY008 pls remove e2e changes

**Review by moremind — APPROVED** [review]

**Pull request merged**